platform/linux-generic/m4/odp_mdev.m4

```diff
@@ -0,0 +1,34 @@
+##########################################################################
+# Enable mdev support
+##########################################################################
+mdev_support=no
```

Comment:
This is fine as a default, but we need:

1. The `configure` script should include this setting (yes/no) in its summary printout of configuration options

2. Travis needs to include appropriate test variants with `--enable-mdev-support=yes`. Otherwise we're just including uncompiled and untested code, which isn't good.

3. Do we need any updates to the validation tests to exercise this code? Those can be a follow-on PR as long as the basics compile.

When I manually run `./configure --enable-mdev-support=yes` I get the following:

```
  CC       pktio/mdev/i40e.lo
pktio/mdev/i40e.c:31:10: fatal error: uapi/net_mdev.h: No such file or directory
 #include <uapi/net_mdev.h>
          ^~~~~~~~~~~~~~~~~
compilation terminated.
Makefile:1234: recipe for target 'pktio/mdev/i40e.lo' failed
make[1]: *** [pktio/mdev/i40e.lo] Error 1
make[1]: Leaving directory '/home/bill/linaro/hubodp/platform/linux-generic'
Makefile:528: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

I realize this PR has external dependencies, but they need to be documented so that they can be provided to permit compilation and testing. We can't include uncompilable code in the repo.

Reply:
This is probably a typo since the required header is located in platform/linux-generic/pktio/uapi_net_mdev.h

Reply:
Correct, this is indeed typo.